Pinta.Core/Extensions/OtherExtensions.cs

```diff
@@ -29,90 +28,91 @@ public static bool In<T> (this T enumeration, params T[] values)
 
 		public static PointI[][] CreatePolygonSet (this BitMask stencil, RectangleD bounds, int translateX, int translateY)
 		{
+			if (stencil.IsEmpty)
+				return Array.Empty<PointI[]> ();
+
 			var polygons = new List<PointI[]> ();
 
-			if (!stencil.IsEmpty) {
-				PointI start = bounds.Location ().ToInt ();
-				var pts = new List<PointI> ();
-				int count = 0;
+			PointI start = bounds.Location ().ToInt ();
+			var pts = new List<PointI> ();
+			int count = 0;
 
-				// find all islands
-				while (true) {
-					bool startFound = false;
+			// find all islands
+			while (true) {
+				bool startFound = false;
 
-					while (true) {
-						if (stencil[start]) {
-							startFound = true;
-							break;
-						}
+				while (true) {
+					if (stencil[start]) {
+						startFound = true;
+						break;
+					}
 
-						++start.X;
+					++start.X;
 
-						if (start.X >= bounds.Right) {
```

Comment:
I find the original version here `if (start.X >= bound.Right)` more readable since it's a short if block so I'd prefer restoring that.
The other changes all look good to me!

Reply:
No problem, I just inverted the if block again